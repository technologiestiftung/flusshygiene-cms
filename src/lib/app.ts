import cors from 'cors';
import errorHandler from 'errorhandler';
import express from 'express';
import * as reactViews from 'express-react-views';
import helmet from 'helmet';
import morgan from 'morgan';
// import Bundler from 'parcel-bundler';
import path from 'path';
import router from './router';

// -------------------------------------------------

const app = express();
app.use(cors());
app.use(helmet());
app.use(morgan('dev'));
if (process.env.NODE_ENV === 'development') {
  app.use(errorHandler());
  // const bundler  = new Bundler(path.resolve(__dirname, '../../src-public/assets/js/index.js'),
  // {
  //   cache: false,
  //   outDir: path.resolve(__dirname, '../../src-public/assets/js/'),
  //   outFile: 'index.bundle.js',
  // });
  // app.use(bundler.middleware());
}

app.use(express.static(path.join(__dirname, '../../public')));

app.set('views', path.join(__dirname, './views/pages'));
app.engine('js', reactViews.createEngine({beautify: true}));
app.set('view engine', 'js');

// app.get('/*', (request, response) => {
//   response.send(`${request.url}`);
// });
app.use(router);

export = app;
