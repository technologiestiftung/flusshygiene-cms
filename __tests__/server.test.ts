import request from 'supertest';
import app from '../src/lib/app';

describe('default testing for server', () => {
  it('should response with 200 on /', async () => {
    expect.assertions(2);
    const response = await request(app).get('/');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });
  // it('should accept post', async () => {
  //   expect.assertions(1);
  //   const response = await request(app)
  //     .post('/submit')
  //     .send({foo: 'bah'});
  //   expect(response.status).toBe(200);
  // });
  // it('should return the passed object', async () => {
  //   expect.assertions(1);
  //   const obj = {foo: 'bah'};
  //   const response = await request(app)
  //     .post('/submit')
  //     .send(obj);
  //   expect(response.text).toEqual('{"code":0,"data":{"foo":"bah"},"errors":""}');
  // });
});
