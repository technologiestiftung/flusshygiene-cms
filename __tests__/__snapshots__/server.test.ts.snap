// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default testing for server should response with 200 on / 1`] = `"/"`;
